File: src/main/java/spring/springserver/domain/community/data/response/CommunityPostResponse.java
package spring.springserver.domain.community.data.response;

import spring.springserver.domain.community.entity.CommunityPost;

import java.time.LocalDateTime;

public record CommunityPostResponse(

        Long id,

        String username,

        String title,

        String content,

        String imageUrl,

        int viewCount,

        LocalDateTime createdAt,

        LocalDateTime updatedAt
) {

    public static CommunityPostResponse from(CommunityPost communityPost) {

        return new CommunityPostResponse(
                communityPost.getId(),
                communityPost.getUsername(),
                communityPost.getTitle(),
                communityPost.getContent(),
                communityPost.getImageUrl(),
                communityPost.getViewCount(),
                communityPost.getCreatedAt(),
                communityPost.getUpdatedAt()
        );
    }
}

File: src/main/java/spring/springserver/domain/community/data/response/CommunityPostResponse.kt
package spring.springserver.domain.community.data.response

import spring.springserver.domain.community.entity.CommunityPost
import java.time.LocalDateTime

data class CommunityPostResponse(
    val id: Long,

    val username: String,

    val title: String,

    val content: String?,

    val fileUrl: String?,

    val viewCount: Int,

    val isEdited: Boolean,

    val commentCount: Long,

    val createdAt: LocalDateTime?,

    val updatedAt: LocalDateTime?,
) {
    companion object {

        fun of(communityPost: CommunityPost, commentCount: Long): CommunityPostResponse {

            return CommunityPostResponse(
                communityPost.getId()!!,
                communityPost.username,
                communityPost.title,
                communityPost.content,
                communityPost.fileUrl,
                communityPost.viewCount,
                communityPost.isEdited,
                commentCount,
                communityPost.getCreatedAt(),
                communityPost.getUpdatedAt(),
            )
        }
    }
}
